utils/vptr/include/virtual_ptr.hpp

```diff
@@ -0,0 +1,510 @@
+#endif  // VIRTUAL_PTR_VERBOSE
+    return retVal;
+  }
+
+  /**
+   * @brief Fuses the given node with the previous nodes in the
+   *        pointer map if they are free
+   *
+   * @param node A reference to the free node to be fused
+   */
+  void fuse_forward(typename pointerMap_t::iterator &node) {
+    while (node != std::prev(m_pointerMap.end())) {
+      // if following node is free
+      // remove it and extend the current node with its size
+      auto fwd_node = std::next(node);
+      if (!fwd_node->second._free) {
+        break;
+      }
+      auto fwd_size = fwd_node->second._size;
+      m_freeList.erase(fwd_node);
+      m_pointerMap.erase(fwd_node);
+
+      node->second._size += fwd_size;
+    }
+  }
+
+  /**
+   * @brief Fuses the given node with the following nodes in the
+   *        pointer map if they are free
+   *
+   * @param node A reference to the free node to be fused
+   */
+  void fuse_backward(typename pointerMap_t::iterator &node) {
+    while (node != m_pointerMap.begin()) {
+      // if previous node is free, extend it
+      // with the size of the current one
+      auto prev_node = std::prev(node);
+      if (!prev_node->second._free) {
+        break;
+      }
+      prev_node->second._size += node->second._size;
+
+      // remove the current node
+      m_freeList.erase(node);
+      m_pointerMap.erase(node);
+
+      // point to the previous node
+      node = prev_node;
+    }
+  }
+
+  /* remove_pointer.
+   * Removes the given pointer from the map.
+   */
+  void remove_pointer(const virtual_pointer_t ptr) {
+    auto node = this->get_node(ptr);
+
+    node->second._free = true;
+    m_freeList.emplace(node);
+
+    // Fuse the node
+    // with free nodes before and after it
+    fuse_forward(node);
+    fuse_backward(node);
+
+    // If after fusing the node is the last one
+    // simply remove it (since it is free)
+    if (node == std::prev(m_pointerMap.end())) {
+      m_freeList.erase(node);
+      m_pointerMap.erase(node);
+    }
+
+#if VIRTUAL_PTR_VERBOSE
+    std::cout << "New list after removing: " << static_cast<long>(ptr)
+              << std::endl;
+    for (auto &n : m_pointerMap) {
+      std::cout << static_cast<long>(n.first) << " { "
+                << n.second._b.get_impl().get() << ", "
+                << ((n.second._free) ? "Freed" : "Usable") << ", "
+                << n.second._b.get_count() << ", " << n.second._size << " }"
+                << std::endl;
+    }
+#endif  // VIRTUAL_PTR_VERBOSE
+  }
+
+  /* count.
+   * Return the number of active pointers (i.e, pointers that
+   * have been malloc but not freed).
+   */
+  size_t count() const {
+    if (!m_pointerMap.empty()) {
+#if VIRTUAL_PTR_VERBOSE
+      std::cout << " Map size " << m_pointerMap.size() << " "
+                << m_freeList.size() << std::endl;
+#endif  // VERBOSE
+      return (m_pointerMap.size() - m_freeList.size());
```

Comment:
I now understand, the original comment relates to checking `if(!m_pointerMap.empty())` earlier in the method. Since if the pointerMap is empty, its size is 0 AND m_freeList is empty too, then we do not need to check if empty and return 0 explicitly.
Fixed: f11cac911e8e2a3c5dbad295096d5c9fa310ae8e